apps/www/pages/anmelden.js

```diff
@@ -11,12 +11,29 @@ import { withDefaultSSR } from '../lib/apollo/helpers'
 import { useMe } from '../lib/context/MeContext'
 
 const SigninPage = () => {
-  const { query } = useRouter()
+  const router = useRouter()
+  const { query } = router
   const { me } = useMe()
   const { t } = useTranslation()
   useEffect(() => {
-    if (me) {
-      window.location = '/'
+    if (me && query?.redirect) {
+      try {
+        const redirectTarget = decodeURIComponent(query.redirect)
+        const redirectUrl = new URL(redirectTarget, window.location.origin)
+        // ensure that the redirect target can only be a relative path
+        if (
+          redirectUrl.hostname === window.location.hostname &&
+          redirectTarget.startsWith('/')
+        ) {
+          router.replace(redirectUrl)
+        } else {
+          router.replace('/')
+        }
+      } catch (e) {
+        console.error(e)
```

Comment:
Again, is there no sensible fallback behavior instead of logging the error? Like doing `router.replace('/')` …